src/api/firebase.ts

```diff
@@ -301,3 +301,44 @@ export async function deleteItem() {
 	 * this function must accept!
 	 */
 }
+
+export function comparePurchaseUrgency(item1: ListItem, item2: ListItem) {
+	const currentDate = new Date();
+
+	// Check for last time item had any activity
+	const daysSinceItemLastActivity = item1.dateLastPurchased
+		? getDaysBetweenDates(currentDate, item1.dateLastPurchased.toDate())
+		: getDaysBetweenDates(currentDate, item1.dateCreated.toDate());
+
+	// Prioritize item1 if current date is past next purchase date
+	if (currentDate > item1.dateNextPurchased.toDate()) {
+		return -1;
+	}
+
+	// De-prioritize an item if its had no activity for more than 60 days
+	if (daysSinceItemLastActivity >= 60) {
+		return 1;
+	}
+	//Compare days until next purchase for item1 and item2
+	//if item1 needs to be purchased sooner, prioritize item1 over item2
+	if (
+		getDaysBetweenDates(currentDate, item1.dateNextPurchased.toDate()) <
+		getDaysBetweenDates(currentDate, item2.dateNextPurchased.toDate())
+	) {
+		return -1;
+
+		//if item2 needs to be purchased sooner, prioritize item2 over item1
+	} else if (
+		getDaysBetweenDates(currentDate, item1.dateNextPurchased.toDate()) >
+		getDaysBetweenDates(currentDate, item2.dateNextPurchased.toDate())
+	) {
+		return 1;
+		//if both items have the same sort order, we sort alphabetically
+	} else {
+		if (item1.name.toLowerCase() < item2.name.toLowerCase()) {
+			return -1;
+		} else {
+			return 1;
```

Comment:
Nice work figuring out this logic!🌟🌟 It's a real brain twister at the start but this is very well done. know in the past others have explored switch functions for a cleaner look and can be faster to compile depending on the use case. Like DJ said above this could also just be tightened up with turning some `else if` statements to `if` statements too.